src/main/java/com/gluonhq/substrate/target/AndroidTargetConfiguration.java

```diff
@@ -70,25 +70,25 @@
 
     public AndroidTargetConfiguration( ProcessPaths paths, InternalProjectConfiguration configuration ) {
         super(paths,configuration);
-        // for now, we need to have an ANDROID_NDK
-        // we will fail fast whenever a method is invoked that uses it (e.g. compile)
-        String sysndk = System.getenv("ANDROID_NDK");
-        if (sysndk != null) {
-            this.ndk = sysndk;
-            Path ldguess = Paths.get(this.ndk, "toolchains", "llvm", "prebuilt", "linux-x86_64", "bin", "ld.lld");
-            if (Files.exists(ldguess)) {
-                ldlld = ldguess;
-            } else {
-                ldlld = null;
-            }
-            Path clangguess = Paths.get(this.ndk, "toolchains", "llvm", "prebuilt", "linux-x86_64", "bin", "clang");
-            clang = Files.exists(clangguess) ? clangguess : null;
-        } else {
-            this.ndk = null;
-            this.ldlld = null;
-            this.clang = null;
+        
+        String tmpSdk = "";
+        String tmpNdk = "";
+
+        try { 
+            tmpSdk = fileDeps.getAndroidSDKPath().toString();
+            tmpNdk = fileDeps.getAndroidNDKPath().toString();
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Don't we want to fail as fast as possible? When an IOException occurs, it probably means that something failed during the automatic download of the android tools (SDK, NDK or additional libraries). The user will most likely get the following error message instead:

> You specified an android ndk, but it doesn't contain /toolchains/llvm/prebuilt/linux-x86_64/bin/ldlld

This is both cryptic and and a bit misleading, because the error might be for instance that the download of the Android SDK failed because of http proxy issues. Or any other actual IO exception that caused the download, unpacking or installing any of the android tools to fail.

Reply:
Should be fixed now